libraries/botbuilder/src/cloudAdapter.ts

```diff
@@ -41,13 +42,25 @@ import {
 const ActivityT = z.custom<Activity>((val) => z.record(z.unknown()).check(val), { message: 'Activity' });
 
 export class CloudAdapter extends CloudAdapterBase implements BotFrameworkHttpAdapter {
+    protected readonly webSocketFactory: NodeWebSocketFactoryBase;
+
     /**
      * Initializes a new instance of the [CloudAdapter](xref:botbuilder:CloudAdapter) class.
      *
      * @param botFrameworkAuthentication Optional [BotFrameworkAuthentication](xref:botframework-connector.BotFrameworkAuthentication) instance
+     * @param webSocketFactory Optional [NodeWebSocketFactoryBase](xref:botframework-streaming.NodeWebSocketFactoryBase) instance
      */
-    constructor(botFrameworkAuthentication: BotFrameworkAuthentication = BotFrameworkAuthenticationFactory.create()) {
+    constructor(
+        botFrameworkAuthentication: BotFrameworkAuthentication = BotFrameworkAuthenticationFactory.create(),
+        webSocketFactory: NodeWebSocketFactoryBase = new NodeWebSocketFactory()
+    ) {
         super(botFrameworkAuthentication);
+
+        if (!webSocketFactory) {
+            throw new TypeError('`webSocketFactory` parameter required');
```

Comment:
If the webSocketFactory is null, should this default to `NodeWebSocketFactory` since this is an optional parameter?